api4/user.go

```diff
@@ -745,6 +754,19 @@ func patchUser(c *Context, w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	// If eMail update is attempted by the currently logged in user, check if correct password was provided
+	if patch.Email != nil && ouser.Email != *patch.Email && c.App.Session.UserId == c.Params.UserId {
+
+		if patch.Password != nil {
+			err = c.App.DoubleCheckPassword(ouser, *patch.Password)
+		}
+
+		if patch.Password == nil || err != nil {
```

Comment:
Yeah, I think that makes it a bit clearer what we're trying to do here. I'd even lean towards combining moving `DoubleCheckPassword` into that if statement since I'm not a fan of reusing `err` in the inner scope
```go
	// If eMail update is attempted by the currently logged in user, check if correct password was provided
	if patch.Email != nil && ouser.Email != *patch.Email && c.App.Session.UserId == c.Params.UserId {
		if patch.Password == nil {
			c.SetInvalidParam("password")
			return
		}

		if err := c.App.DoubleCheckPassword(ouser, *patch.Password); err != nil {
			c.SetInvalidParam("password")
			return
		}
	}
```

Reply:
Ya I think that is clearer.